lib/custom/src/MShop/Service/Provider/Payment/PaypalPlus.php

```diff
@@ -0,0 +1,354 @@
+			| \Aimeos\MShop\Order\Item\Base\Base::PARTS_PRODUCT
+			| \Aimeos\MShop\Order\Item\Base\Base::PARTS_ADDRESS;
+
+		$base = $this->getOrderBase( $order->getBaseId(), $parts );
+		$data = $this->getData( $base, $order->getId(), $params );
+		$urls = $this->getPaymentUrls();
+		$data['cancelUrl'] = $this->getConfigValue( 'cancelUrl', '' ) ;
+
+		
+		try
+		{
+			$response = $this->sendRequest( $order, $data );
+			$isSuccessful = $response->isSuccessful()  ;
+
+			/*
+			if($isSuccessful){
+				$php_a = $response->getData()['links']['1']['href'] ;
+				echo ($php_a  ) ;
+
+				echo "<script>
+				var approval_ur = '$php_a'; 
+			//    alert(approval_ur);
+				</script>" ;
+
+				echo '
+				<script src="https://www.paypalobjects.com/webstatic/ppplus/ppplus.min.js" type="text/javascript">
+				</script>
+				<div id="ppplus">
+				</div>
+				<script type="application/javascript">
+
+				//  alert(approval_ur) ;
+					var ppp = PAYPAL.apps.PPP({
+					"approvalUrl": approval_ur,
+					"placeholder": "ppplus",
+					"mode": "sandbox",
+					"country": "DE"
+					});
+					setTimeout(function(){ alert("Hello"); }, 3000);
+
+				//	alert(approval_ur) ;
+				</script>
+				
+				';
+			//	die() ;
+			}
+		//	*/
+
+
+			if( $isSuccessful )
+			{
+				$this->setOrderData( $order, ['Transaction' => $response->getTransactionReference()] );
+				$this->saveRepayData( $response, $base->getCustomerId() );
+
+				$status = $this->getValue( 'authorize', false ) ? Status::PAY_AUTHORIZED : Status::PAY_RECEIVED;
+				$this->saveOrder( $order->setPaymentStatus( $status ) );
+			}
+			elseif( $response->isRedirect() )
+			{
+				$this->setOrderData( $order, ['Transaction' => $response->getTransactionReference()] );
+				return $this->getRedirectForm( $response );
+			}
+			else
+			{
+				$this->saveOrder( $order->setPaymentStatus( Status::PAY_REFUSED ) );
+				throw new \Aimeos\MShop\Service\Exception( $response->getMessage() );
+			}
+		}
+		catch( \Exception $e )
+		{
+			throw new \Aimeos\MShop\Service\Exception( $e->getMessage() );
+		} 
+		$php_a = $response->getData()['links']['1']['href'] ;
+	//	$url = $this->getConfigValue( 'cancelUrl', '' );
+	//	 dd($url) ;
+		return new \Aimeos\MShop\Common\Helper\Form\Standard( $urls['returnUrl'] ?? '', 'POST', [], false, $this->getPayPalPlusJs($php_a) );
+
+	}
+
+	protected function getPayPalPlusJs($php_a) : string
+	{
+		echo '
+		
+		<script src="https://www.paypalobjects.com/webstatic/ppplus/ppplus.min.js" type="text/javascript">
+		</script>
+		
+		<script>
+		var approval_ur = \'' . $php_a . '\'; 
+		//	setTimeout(function(){ alert("Done"); }, 1000);
+		//	alert(approval_ur);
+
+		</script> 
+ 
+
+		<div id="ppplus">
+		</div>
+		<script type="application/javascript">
+		//	setTimeout(function(){ alert("Done"); }, 3000);
+
+		//  alert(approval_ur) ;
+			var ppp = PAYPAL.apps.PPP({
+			"approvalUrl": approval_ur,
+			"placeholder": "ppplus",
+			"mode": "sandbox",
```

Comment:
This should be made configurable depending on the `$this->getConfigValue( 'testmode' )`

Reply:
thanks for note ,I have updated my code